app/javascript/components/patient/laboratory/LaboratoryModal.js

```diff
@@ -3,61 +3,130 @@ import { PropTypes } from 'prop-types';
 import { Alert, Button, Col, Form, Modal, Row } from 'react-bootstrap';
 import moment from 'moment';
 import ReactTooltip from 'react-tooltip';
-
+import * as yup from 'yup';
+import _ from 'lodash';
 import DateInput from '../../util/DateInput';
 
 class LaboratoryModal extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      lab_type: this.props.lab?.lab_type || '',
-      specimen_collection: this.props.lab?.specimen_collection,
-      report: this.props.lab?.report,
-      result: this.props.lab?.result || (this.props.onlyPositiveResult ? 'positive' : ''),
-      reportInvalid: false,
+      lab_type: this.props.currentLabData?.lab_type || '',
+      specimen_collection: this.props.currentLabData?.specimen_collection,
+      report: this.props.currentLabData?.report,
+      result: this.props.currentLabData?.result || (this.props.firstPositiveLab ? 'positive' : ''),
+      symptom_onset: null,
+      loading: false,
+      errors: {},
     };
+    this.updateValidations();
+  }
+
+  updateValidations() {
+    let reportDate = '2020-01-01';
+    let reportMessage = 'Report Date must fall after January 1, 2020.';
+    if (this.state.report && this.state.specimen_collection) {
+      // if both dates are set, ensure the report date falls after the specimen collection date
+      reportDate = this.state.specimen_collection;
+      reportMessage = 'Report Date cannot be before Specimen Collection Date.';
+    }
+
+    if (this.props.firstPositiveLab) {
+      schema = yup.object().shape({
+        lab_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.').nullable(),
+        specimen_collection: yup
+          .date('Date must correspond to the "mm/dd/yyyy" format.')
+          .required('Please enter Specimen Collection Date.')
+          .min(moment('2020-01-01'), 'Specimen Collection Date must fall after January 1, 2020.')
+          .max(moment(), 'Specimen Collection Date can not be in the future.')
+          .nullable(),
+        report: yup
+          .date('Date must correspond to the "mm/dd/yyyy" format.')
+          .min(moment(reportDate), reportMessage)
+          .max(moment(), 'Report Date can not be in the future.')
+          .nullable(),
+        result: yup.string().required('Please select a lab result.').max(200, 'Max length exceeded, please limit to 200 characters.').nullable(),
+      });
+    } else {
+      schema = yup.object().shape({
+        lab_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.').nullable(),
+        specimen_collection: yup
+          .date('Date must correspond to the "mm/dd/yyyy" format.')
+          .min(moment('2020-01-01'), 'Specimen Collection Date must fall after January 1, 2020.')
+          .max(moment(), 'Specimen Collection Date can not be in the future.')
```

Comment:
can not --> cannot

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1184/commits/5e88210577f5c4cbdf5b1da7d3deb72b2a8f85a2